lib/src/main/java/space/taran/arklib/data/preview/Preview.kt

```diff
@@ -29,27 +30,35 @@ data class Preview(
 
     // we don't have fullscreen preview for
     // some kinds of resources, e.g. Image or PlainText
-    val onlyThumbnail: Boolean) {
+    val onlyThumbnail: Boolean
+) {
 
     internal companion object {
         const val THUMBNAIL_SIZE = 128
         const val COMPRESSION_QUALITY = 100
 
-        fun downscale(bitmap: Bitmap): Bitmap =
-            downscale(Glide.with(app).asBitmap().load(bitmap))
-
-        fun downscale(builder: RequestBuilder<Bitmap>): Bitmap =
-            builder
-                .apply(RequestOptions()
-                    .downsample(DownsampleStrategy.CENTER_INSIDE)
-                    .override(THUMBNAIL_SIZE)
-                    .fitCenter()
-                    .diskCacheStrategy(DiskCacheStrategy.NONE)
-                    .skipMemoryCache(true)
+        /**
+         * See [ImageRequest.Builder.data] for supported data types
+         * Note: pass [java.io.File] instead of [java.nio.file.Path]
+         */
+        suspend fun downscale(resource: Path, data: Any): Bitmap {
+            val request = ImageRequest.Builder(app)
+                .size(THUMBNAIL_SIZE)
+                .precision(Precision.EXACT)
+                .scale(Scale.FIT)
+                .data(data)
+                .listener(
+                    onError = { _, result ->
+                        Log.d(
+                            LOG_PREFIX,
+                            "Failed to downscale preview for $resource because ${result.throwable}"
+                        )
+                    },
                 )
-                .addListener(ImageUtils.glideExceptionListener<Bitmap>())
-                .submit()
-                .get()
+                .build()
+
+            return ImageUtils.arkImageLoader.execute(request).drawable!!.toBitmap()
```

Comment:
What if the load request returns a null `drawable`? Crash?
Let's not use the non-null assertion (!!) here, handle the null result as well.

Reply:
Actually we should always successfully downscale preview, but yeah, it’s better to wrap it in result